### PR TITLE
chore: Drop v1beta1 references

### PR DESCRIPTION
**Pull request by jigisha620**
Fixes #N/A 

**Description**
Dropping v1beta1 references for next release.

**How was this change tested?**

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -114,3 +114,5 @@ require (
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.4.1 // indirect
 )
+
+replace sigs.k8s.io/karpenter => github.com/jigisha620/karpenter v0.0.0-20240827203829-de203981717b
```
Comment (jigisha620) [inline go.mod:118]: Will drop this once upstream PR is merged.

**Review by jigisha620 — COMMENTED** [review]
/karpenter snapshot

**Review by engedaam — COMMENTED** [review]
/lgtm

**Review by jonathan-innis — DISMISSED** [review]
LGTM 🚀

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Comment by KalenWessel** [comment]
What release can we expect this fix to be in?

**Pull request merged**